File: HW1/3/tf_06_test.go
package main

import (
	"reflect"
	"testing"
)

func TestReadFile(t *testing.T) {
	want := "This is a test."
	str := ReadFile("../test.txt")
	if str != want {
		t.Errorf("ReadFile() = %v, want %v", str, want)
	}
}

func TestFilterCharsAndNormalize(t *testing.T) {
	want := "this is a test "
	str := FilterCharsAndNormalize("This is a test.")
	if str != want {
		t.Errorf("FilterCharsAndNormalize() = %v, want %v", str, want)
	}
}

func TestScan(t *testing.T) {
	want := []string{"this", "is", "a", "test"}
	wordList := Scan("this is a test")
	if !reflect.DeepEqual(wordList, want) {
		t.Errorf("Scan() = %v, want %v", wordList, want)
	}
}

func TestRemoveStopWords(t *testing.T) {
	wordList := []string{"work", "is", "hard", "a", "work", "pays", "off"}
	want := []string{"work", "hard", "work", "pays"}
	newWordList := RemoveStopWords(wordList)("../../stop_words.txt")
	// ("../../stop_words.txt")
	if !reflect.DeepEqual(newWordList, want) {
		t.Errorf("RemoveStopWords() = %v, want %v", newWordList, want)
	}
}

func TestFrequencies(t *testing.T) {
	wordList := []string{"work", "work", "is", "hard", "a", "a", "work", "pays", "off"}
	want := map[string]int{"work": 3, "is": 1, "hard": 1, "a": 2, "pays": 1, "off": 1}
	wordFreqs := Frequencies(wordList)
	if !reflect.DeepEqual(wordFreqs, want) {
		t.Errorf("Frequencies() = %v, want %v", wordFreqs, want)
	}
}

func TestSort(t *testing.T) {
	wordFreqs := map[string]int{"work": 3, "is": 1, "hard": 1, "a": 2, "pays": 1, "off": 1}
	want := []pair{{"work", 3}, {"a", 2}, {"is", 1}, {"hard", 1}, {"pays", 1}, {"off", 1}}
	sortedWordFreqs := Sort(wordFreqs)
	if !reflect.DeepEqual(sortedWordFreqs, want) {
		t.Errorf("Sort() = %v, want %v", sortedWordFreqs, want)
	}
}

File: HW1/1/tf_06_test.py
import unittest
import tf_06

class TestTf06(unittest.TestCase):
    def test_read_file(self):
        self.assertEqual(tf_06.read_file('../test.txt'), 'This is a test.')

    def test_filter_chars_and_normalize(self):
        self.assertEqual(tf_06.filter_chars_and_normalize('This is a test.'), 'this is a test ')

    def test_scan(self):
        self.assertEqual(tf_06.scan('this is a test '), ['this', 'is', 'a', 'test'])

    def test_remove_stop_words(self):
        word_list = ['work', 'is', 'hard', 'a', 'work', 'pays', 'off']
        result = tf_06.remove_stop_words(word_list)('../../stop_words.txt')
        self.assertEqual(result, ['work', 'hard', 'work', 'pays'])

    def test_frequencies(self):
        word_list = ['work', 'hard', 'work', 'pays']
        self.assertEqual(tf_06.frequencies(word_list), {'work': 2, 'hard': 1, 'pays': 1})

    def test_sort(self):
        word_freq = {'work': 3, 'hard': 1, 'pays': 2}
        self.assertEqual(tf_06.sort(word_freq), [('work', 3), ('pays', 2), ('hard', 1)])

if __name__ == '__main__':
    unittest.main()
